### PR TITLE
Block external jsonapi access + add new ingress

### PR DESCRIPTION
**Pull request by leonk**
### Context

> Does this issue have a Trello card?

https://trello.com/c/6vgxPaTU/492-only-allow-access-to-content-from-within-cloud-platform-network

### Intent

This PR adds a nginx configuration snippet that denies external access to /jsonapi routes.
After this update, only requests within cloud platform will be accepted through to any path starting with /jsonapi.

It also adds a new ingress + hostname, that is allowlisted to the MoJ network.  This does not include the nginx config to block `/jsonapi` routes, so can be used for local development + debugging.
The hostnames for each environment are:
- jsonapi-cms-prisoner-content-hub-development.apps.live.cloud-platform.service.justice.gov.uk
- jsonapi-cms-prisoner-content-hub-staging.apps.live-1.cloud-platform.service.justice.gov.uk
- jsonapi-cms-prisoner-content-hub-production.apps.live-1.cloud-platform.service.justice.gov.uk

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

**Comment by leonk** [comment]
I'm not quite sure which is the better approach, this or https://github.com/ministryofjustice/prisoner-content-hub-backend/pull/314.

This one has a lot more config, and adds an extra hostname that we need to manage.
The other PR is a lot more basic, but the development workflow (of setting up port forwarding) is not quite as nice.

File: helm_deploy/prisoner-content-hub-backend/values.production.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ ingress:
     - host: manage.content-hub.prisoner.service.justice.gov.uk
       cert_secret: prisoner-content-hub-cms-certificate
     - host: cms-prisoner-content-hub-production.apps.live-1.cloud-platform.service.justice.gov.uk
+  jsonapi:
+    annotations:
+      external-dns.alpha.kubernetes.io/set-identifier: prisoner-content-hub-backend-jsonapi-prisoner-content-hub-production-blue
+    hosts:
+      - host: jsonapi-cms-prisoner-content-hub-production.apps.live-1.cloud-platform.service.justice.gov.uk
```
Comment (leonk) [inline helm_deploy/prisoner-content-hub-backend/values.production.yaml:12]: Do we need a host for production?  Could remove this potentially, but might be nice to be able to debug issues.

File: helm_deploy/prisoner-content-hub-backend/templates/_helpers.tpl [hunk 2]
```diff
@@ -81,3 +81,24 @@ Create trusted host pattern
...
+*/}}
+{{- define "prisoner-content-hub-backend.trustedHostsJsonApi" -}}
+{{- with (first .Values.ingress.jsonapi.hosts) -}}
+^{{ (.host | replace "." "\\.") }}$
+{{- end }}
+{{- range (slice .Values.ingress.jsonapi.hosts 1) -}}
+|^{{ (.host | replace "." "\\.")}}$
+{{- end }}
+{{- printf "|^%s\\.%s\\.svc\\.cluster\\.local$" (include "prisoner-content-hub-backend.fullname" .) .Release.Namespace }}
+{{- end }}
```
Comment (andrewrlee) [inline helm_deploy/prisoner-content-hub-backend/templates/_helpers.tpl:96]: as discussed, would be nice to tidy this up but happy to leave it for now

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**